install/kubernetes/cilium/README.md

```diff
@@ -155,6 +155,9 @@ contributors across the globe, there is almost always someone available to help.
 | containerRuntime.integration | string | `"none"` | Enables specific integrations for container runtimes. Supported values: - containerd - crio - docker - none - auto (automatically detect the container runtime) |
 | customCalls | object | `{"enabled":false}` | Tail call hooks for custom eBPF programs. |
 | customCalls.enabled | bool | `false` | Enable tail call hooks for custom eBPF programs. |
+| daemon.allowedConfigOverrides | string | `nil` | allowedConfigOverrides is a list of config-map keys that can be overridden. That is to say, if this value is set, config sources (excepting the first one) can only override keys in this list.  This takes precedence over blockedConfigOverrides.  By default, all keys may be overridden. To disable overrides, set this to "none" or change the configSources variable. |
+| daemon.blockedConfigOverrides | string | `nil` | blockedConfigOverrides is a list of config-map keys that may not be overridden. In other words, if any of these keys appear in a configuration source excepting the first one, they will be ignored  This is ignored if allowedConfigOverrides is set.  By default, all keys may be overridden. |
+| daemon.configSources | string | `nil` | Configure a custom list of possible configuration override sources The default is "config-map:cilium-config,cilium-node-config" Possible entries are   - "configmap:<namespace>/<name>" -- a ConfigMap with name <name>, optionally at a namespace.   - "cilium-node-config:<namespace>/<name>" --  any or all CiliumNodeConfigurations. If <name> is specified,                                                 then only that one is considered. Otherwise, all CNCs in the given namespace                                                 that match the LabelSelector are used. If <namespace> is not specified, it                                                 defaults to the Cilium namespace.   - "node" -- Any annotations on the Node object of the form "io.cilium.config/<KEY> = <VALUE>"  By default, node annotations are not consumed. As configuration overrides have security implications, you should review which accounts may annotate Node objects first.  Note that this value should be a comma-separated string. |
```

Comment:
Hmm, formatting didn't turn out super great here. Broadly speaking I think that the explanation for this should probably live somewhere dedicated rather than in the configmap and generated to here, because this just ends up in a table column so it's gonna be quite hard to read.

Should we have a new page for this feature from the following location to explain how this works? https://docs.cilium.io/en/latest/configuration/

Reply:
Possible other options here: 
- We can leave this as-is, maybe ideally improve the tool that generates this output so that the giant empty spaces are collapsed to 1-2 spaces.
- Move the explanation to docs page, then from the values.yaml and here we remove the explanation (or link to the corresponding docs page)
- Any other ideas?

Reply:
Cleaned up the formatting a bit, #22576 for more docs.